synthesizer/process/src/stack/evaluate.rs

```diff
@@ -82,6 +82,8 @@ impl<N: Network> StackEvaluate<N> for Stack<N> {
                     Operand::Caller => Ok(Value::Plaintext(Plaintext::from(Literal::Address(registers.caller()?)))),
                     // If the operand is the block height, throw an error.
                     Operand::BlockHeight => bail!("Cannot retrieve the block height from a closure scope."),
+                    // If the operand is the network id, throw an error.
+                    Operand::NetworkID => bail!("Cannot retrieve the network ID from a closure scope."),
```

Comment:
What is the reason why this shouldn't be available in evaluation?
Same for below.

Reply:
Discussed offline. Excluding in execution and evaluation does not prevent the desired use case as you can directly pass in a `u32` and check it against the network in a finalize block.